services/dynamic-sidecar/tests/unit/test_api_containers.py

```diff
@@ -32,10 +29,89 @@
 from simcore_service_dynamic_sidecar.core.validation import parse_compose_spec
 from simcore_service_dynamic_sidecar.models.shared_store import SharedStore
 
-ContainerTimes = namedtuple("ContainerTimes", "created, started_at, finished_at")
+WAIT_FOR_DIRECTORY_WATCHER: Final[float] = 0.1
+FAST_POLLING_INTERVAL: Final[float] = 0.1
 
-DEFAULT_COMMAND_TIMEOUT = 5
-WAIT_FOR_DIRECTORY_WATCHER = 0.1
+# UTILS
+
+
+def _create_network_aliases(network_name: str) -> list[str]:
+    return [f"alias_{i}_{network_name}" for i in range(10)]
+
+
+async def _assert_enable_directory_watcher(client: TestClient) -> None:
+    response = await client.patch(
+        f"/{API_VTAG}/containers/directory-watcher", json=dict(is_enabled=True)
+    )
+    assert response.status_code == status.HTTP_204_NO_CONTENT, response.text
+    assert response.text == ""
+
+
+async def _assert_disable_directory_watcher(client: TestClient) -> None:
+    response = await client.patch(
+        f"/{API_VTAG}/containers/directory-watcher", json=dict(is_enabled=False)
+    )
+    assert response.status_code == status.HTTP_204_NO_CONTENT, response.text
+    assert response.text == ""
+
+
+async def _start_containers(client: TestClient, compose_spec: str) -> list[str]:
+    # start containers
+    response = await client.post(
+        f"/{API_VTAG}/containers/tasks", json={"docker_compose_yaml": compose_spec}
+    )
+    assert response.status_code == status.HTTP_202_ACCEPTED, response.text
+    task_id: TaskId = response.json()
+
+    wait_task_completion = True
+    while wait_task_completion:
+        response = await client.get(f"/task/{task_id}")
+        assert response.status_code == status.HTTP_200_OK
+        task_status = response.json()
+        wait_task_completion = not task_status["done"]
+        await asyncio.sleep(FAST_POLLING_INTERVAL)
+
+    response = await client.get(f"/task/{task_id}/result")
+    assert response.status_code == status.HTTP_200_OK
+    result_response = response.json()
+    assert result_response["error"] is None
+    response_containers = result_response["result"]
+
+    shared_store: SharedStore = client.application.state.shared_store
+    container_names = shared_store.container_names
+    assert response_containers == container_names
+
+    return container_names
+
+
+async def _docker_ps_a_container_names() -> list[str]:
+    command = 'docker ps -a --format "{{.Names}}"'
```

Comment:
minor: I don't get why you do not use aiodocker for these things.

Reply:
added note to replace this